src/HealthChecks.MySql/MySqlHealthCheck.cs

```diff
@@ -8,24 +8,38 @@ namespace HealthChecks.MySql;
 /// </summary>
 public class MySqlHealthCheck : IHealthCheck
 {
-    private readonly string _connectionString;
+    private readonly MySqlHealthCheckOptions _options;
 
-    public MySqlHealthCheck(string connectionString)
+    public MySqlHealthCheck(MySqlHealthCheckOptions options)
     {
-        _connectionString = Guard.ThrowIfNull(connectionString);
+        Guard.ThrowIfNull(options.ConnectionString, true);
+        Guard.ThrowIfNull(options.CommandText, true);
+        _options = options;
     }
 
     /// <inheritdoc />
     public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
     {
         try
         {
-            using var connection = new MySqlConnection(_connectionString);
+            using var connection = new MySqlConnection(_options.ConnectionString);

```

Comment:
You forgot to call `MySqlHealthCheckOptions.Configure` here.

Reply:
As mentioned above, I think you're missing something like this:

```csharp
_options?.Configure(connection);
```

Reply:
Done!